packages/admin-panel/src/table/actions.js

```diff
@@ -90,7 +90,7 @@ const refreshDataWithDebounce = debounce(
     const sortString = JSON.stringify(sortObjects);
 
     // Set up columns
-    const columnSources = columns.map(column => column.source);
+    const columnSources = [...columns.map(column => column.source).filter(source => source), 'id'];
```

Comment:
```suggestion
    const columnSources = columns.map(column => column.source).filter(source => source).push('id');
```

This reduces complexity from linear to constant time (as long as we don’t actually need the deep copy) 🏎️